### DISCUSSION TITLE
Importing / migrating from sqlite setup

### DISCUSSION THREAD

**Discussion by dhow** [shlinkio/discussions#132] [Show and tell]
I'd like to share my experience migrating my sqlite setup on Docker to another Docker container running Maria DB setup. 

Originally I was looking for ways to export sqlite into an SQL file and imported into Maria DB via phpMyAdmin but the import process complained with a lot of errors. This got me worried that even if I fixed all bugs in the SQL file there could be records / tables that won't be recognized when read on a Maria DB installation, so I abandoned this method. 

For the second try, I created a new Docker container with Maria DB setup, but I used [the official URL import tool](https://shlink.io/documentation/advanced/import-short-urls/) to import URLs from existing Shlink setup. The import process worked flawlessly, all I needed to do was to run the import command and map traffic over to the new instance running Maria DB. 

There seems to be another way to migrate [using the update command](https://shlink.io/documentation/update-your-instance/), however, it didn't work as expected (or I just didn't know how to make it work -- or maybe it's not supposed to work as a DB migration tool.) 

Good luck to anyone migrating from sqlite!

**Comment by acelaya** [comment]
Thanks for sharing!

The import tool is the official way to migrate to a different database engine.

The update tool is only meant to be used to upgrade to newer versions of Shlink without changing the setup.

**Reply by dhow** [reply]
Thansk @acelaya! To tell the truth, I read your post about a trick to make persistent storage for sqlite setup, which was what I wanted to do given my use case / scope with Shlink, but I decided to migrate it to Maria DB. The decision was the right way because the import process just worked and the performance is not much different (with my very few URLs.) :)
